# DP: Carried over from glibc2.2, need to figure out why it's here.
# DP: (Probably has something to do with fakeroot)
# DP: Date: (Updated 2005-01-02 gotom)

--- Makeconfig	15 Dec 2004 18:51:47 -0000	1.302
+++ Makeconfig	2 Jan 2005 08:36:24 -0000
@@ -541,10 +541,11 @@
 $(subst $(empty) ,:,$(strip $(patsubst -Wl$(comma)-rpath-link=%, %,\
 				       $(filter -Wl$(comma)-rpath-link=%,\
 						$(sysdep-LDFLAGS)))))
+ld_envlib_path = $(shell test x$$LD_LIBRARY_PATH = x || echo -n :$$LD_LIBRARY_PATH)
 run-program-prefix = $(if $(filter $(notdir $(built-program-file)),\
 				   $(tests-static) $(xtests-static)),, \
 			  $(elf-objpfx)$(rtld-installed-name) \
-			  --library-path $(rpath-link)$(patsubst %,:%,$(sysdep-library-path)))
+			  --library-path $(rpath-link)$(patsubst %,:%,$(sysdep-library-path))$(ld_envlib_path))
 else
 run-program-prefix =
 endif
